data_managers/data_manager_picard_index_builder/data_manager/picard_index_builder.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+#Dan Blankenberg
+
+import sys
+import os
+import optparse
+import subprocess
+
+from json import loads, dumps
+
+
+DEFAULT_DATA_TABLE_NAME = "picard_indexes"
+
+def get_id_name( params, dbkey, fasta_description=None):
+    #TODO: ensure sequence_id is unique and does not already appear in location file
+    sequence_id = params['param_dict']['sequence_id']
+    if not sequence_id:
+        sequence_id = dbkey
+    
+    sequence_name = params['param_dict']['sequence_name']
+    if not sequence_name:
+        sequence_name = fasta_description
+        if not sequence_name:
+            sequence_name = dbkey
+    return sequence_id, sequence_name
+
+def build_picard_index( data_manager_dict, fasta_filename, params, target_directory, dbkey, sequence_id, sequence_name, data_table_name=DEFAULT_DATA_TABLE_NAME ):
+    #TODO: allow multiple FASTA input files
+    fasta_base_name = os.path.split( fasta_filename )[-1]
+    prefix = os.path.splitext( fasta_base_name )[0]
+    sym_linked_fasta_filename = os.path.join( target_directory, fasta_base_name )
+    os.symlink( fasta_filename, sym_linked_fasta_filename )
+    args = [ 'java', '-Xmx4G', '-jar', os.path.join( os.environ[ 'JAVA_JAR_PATH' ], 'CreateSequenceDictionary.jar' ), 'R=%s' % sym_linked_fasta_filename, 'O=%s.dict' % prefix ]
```

Comment:
@blankenberg is it save to hard code these values? Alternatively, we could set `-Xmx` vie `job_conf.xml`.